### DISCUSSION TITLE
I am trying to pass in a function from java as a parameter to a c++ function and I keep getting error: implicit instantiation of undefined template 'std::function<void ()>'

### DISCUSSION THREAD

**Discussion by sungho-park-23** [bytedeco/javacpp#722] [Q&A]
I keep seeing this when I try to build on Maven: 
```
error: implicit instantiation of undefined template 'std::function<void ()>'
        ptr->testCallback(ptr0, *ptr1);
```

I believe I may be doing something wrong, but what I hope to do is something like the example in https://github.com/bytedeco/javacpp#creating-callback-functions -- where there is `public static native void sort(IntPointer first, IntPointer last, @ByVal Callback compare); ` and try to call sort from java side.

Can someone help me figure out the correct way to do this?

Code looks like
C++ side:
```
class TestLib {
  public:
     TestLib() {};
     void testCallback(std::function<void()> callback) {
           callback();
     }; 
}
```

java side:
```
public class Test {
  public static class TestLib extends Pointer {
    static {
      Loader.load();
    }
    public TestLib() {
      allocate();
    }
    private native void allocate();
    public static class Callback extends FunctionPointer {
        static {
           Loader.load();
        }
      public Callback() {
        allocate();
      }
      private native void allocate();
      public @Name("Callback") void call() throws Exception {
        System.out.println("Callback from java");
      }
    }
     
    public native void testCallback(String a, @ByRef Callback callback);
  }
  
   public static void main(String[] args) {
       testCallback("test", new TestLib.Callback());
   }
}
```

**Comment by saudet** [comment]
That probably means you forgot to `#include <functional>`

**Reply by sungho-park-23** [reply]
I thought this was the issue too, but I find that adding `#include <functional>` does not solve the error for me.

I found that it works when I switch to `void (*func)()` instead of `std::function` though. Perhaps do I need to include `@Namespace("std")`?

**Reply by sungho-park-23** [reply]
namespace doesn't solve the issue
